core/src/main/java/cz/xtf/core/openshift/OpenShiftWaiters.java

```diff
@@ -56,29 +56,31 @@ public Waiter hasBuildCompleted(String buildConfigName) {
                 .map(Build::getMetadata).map(ObjectMeta::getName)
                 .map(openShift::getBuild).map(Build::getStatus).map(BuildStatus::getPhase)
                 .orElse(null);
-        String reason = "Waiting for completion of latest build " + buildConfigName;
+        final String reason = "Waiting for completion of latest build " + buildConfigName;
 
-        return new SupplierWaiter<>(supplier, "Complete"::equals, "Failed"::equals, TimeUnit.MINUTES, 10, reason)
-                .logPoint(Waiter.LogPoint.BOTH)
-                .failFast(failFast)
-                .interval(5_000);
+        return new SupplierWaiter<>(supplier, "Complete"::equals, "Failed"::equals, TimeUnit.MINUTES, WaitingConfig.timeout(),
```

Comment:
Current default timeout is 180s (3 min) which is much less then 10min. In case we increase `xtf.waiting.timeout` then other timeouts (not just for the build) will take much longer. We might consinder to create timeout just for the build completion with 10 min default and allow to reconfigure. 

cc-ing @mchoma

Reply:
Thanks @mnovak1

> Current default timeout is 180s (3 min) which is much less than 10min.

Ok, but that is just the default (or better to say a fallback value for it).But I wouldn't change it. Users should be aware of what methods use timeout and provide their default accordingly, if they want something more than 3 mins.
WDYT?

Reply:
@mnovak1 I agree with you. I was not sure if we are ok with introducing new properties. But if you are ok we can do it like that. We have discussed that with @fabiobrz and he will change accordingly - introduce new property `xtf.waiting.build.timeout` especially for this usage with default 10 minutes.

Reply:
Hi @mnovak1 - As a second thought and after some discussion with @mchoma I think get your point is that the library itself relies on the hardcoded business logic, i.e. internal calls assume that this method will have a timeout of 10 mins, and the default fallback of 3 mins could harm those calls. 
IMHO this is actually something that could be improved because the whole point of this PR is that those calls timeouts should be changed into something depending on configuration.

So I'll move on by following your suggestion and not touch this method at all, while I'll create a new one. 
Maybe the existing method should be named somehow differently to include the "hardcoded Timeout" concept.

Reply:
Purpose is to allow to set timeout for builds. Now the timeout was hardcoded to 10 min which is safe value for most of the builds. By changing it to `xtf.waiting.timeout` which is 3 min by default, this timeout will be considerably shortened and it might cause timeout failures in consuming test suites as 3min is really not much time for builds.  

Thus I think we need to add another property to configure it.

Reply:
> I think get your point is that the library itself relies on the hardcoded business logic, i.e. internal calls assume that this method will have a timeout of 10 mins

IIUC No, I think this is not happening.

I would vote now for introducing `xtf.waiting.build.timeout` with default 10 minutes and use it in this particular case.

New method could help to avoid need to introduce new property, but that would mean all callers should change to apply this change.

Reply:
Ok, thanks @mnovak1 and @mchoma for your feedback, I am going to update the PR by following your suggestions.